src/box/xrow_update_array.c

```diff
@@ -310,6 +310,36 @@ xrow_update_array_store(struct xrow_update_field *field,
 	return out - out_begin;
 }
 
+/**
+ * Helper function that appends nils in the end so that op will insert
+ * without gaps
+ */
+static int
+xrow_update_array_append_nils(struct xrow_update_field *field,
+			      struct xrow_update_op *op)
+{
+	struct xrow_update_rope *rope = field->array.rope;
+	uint32_t size = xrow_update_rope_size(rope);
+	if (op->field_no < 0 || (uint32_t)op->field_no <= size)
+		return 0;
+	uint32_t nil_count = op->field_no - size;
+	struct xrow_update_array_item *item =
+		(struct xrow_update_array_item *)
+		xrow_update_alloc(rope->ctx, sizeof(*item));
+	if (item == NULL)
+		return -1;
+	assert(mp_sizeof_nil() == 1);
+	char *item_data = (char *) region_alloc(rope->ctx, nil_count);
```

Comment:
In the new code for unary operators (including cast) we don't use whitespace after them.